ddtrace/contrib/asgi/middleware.py

```diff
@@ -106,7 +106,8 @@ async def __call__(self, scope, receive, send):
             if context.trace_id:
                 self.tracer.context_provider.activate(context)
 
-        resource = "{} {}".format(scope["method"], scope["path"])
+        resource = "{} {}".format(scope["method"], ("resource" in scope and scope["resource"]) or scope["path"])
```

Comment:
nit: I'd prefer if we expanded this into a normal if statement to be more clear.

```python
if "resource" in scope:
    resource = "{} {}".format(scope["method"], scope["resource"])
else:
    resource = "{} {}".format(scope["method"], scope["path"])
```

Reply:
Also let's use `"_dd_resource"` as the entry in scope so that we don't collide with any other code that might use `"resource"`.